zerver/openapi/openapi.py

```diff
@@ -104,6 +109,16 @@ def regex_keys(self) -> Dict[str, str]:
         assert(len(self.regex_dict) > 0)
         return self.regex_dict
 
+    def core_validator(self) -> Any:
+        """Reload the OpenAPI file if it has been modified after the last time
+        it was read, and then return the parsed data.
+        """
+        last_modified = os.path.getmtime(self.path)
+        # Using != rather than < to cover the corner case of users placing an
+        # earlier version than the current one
+        if self.last_update != last_modified:
+            self.reload()
+        return self.core_data
```

Comment:
Can you make adding this be in its own commit that explains why we need this added function?

Reply:
Done